### PR TITLE
Remove implementation of test hooks

### PR DESCRIPTION
**Pull request by jubianchi**
This will help us make the phpunit extension work.

PHPUnit also defines hooks which are in conflict with atoum ones. We make the method "virtual" by removing their implementations and using a runtime check to see if they are implemented in test classes.

### CODE REVIEW

**Comment by Grummfy** [comment]
seems good.

File: classes/test.php [hunk 1]
```diff
@@ -1462,20 +1448,40 @@ public function errorHandler($errno, $errstr, $errfile, $errline)
         return $doNotCallDefaultErrorHandler;
     }
 
-    public function setUp()
+    private function callSetUp()
```
Comment (Hywan) [inline classes/test.php]: These new `call*` methods must be protected to be overriden in the `phpunit-extension`. See my comment to have more explanations.

**Comment by Hywan** [comment]
Based on the [PHPUnit's documentation](https://phpunit.de/manual/current/en/fixtures.html#fixtures.more-setup-than-teardown):

> The setUp() and tearDown() template methods are run once for each test method (and on fresh instances) of the test case class.
> 
> In addition, the setUpBeforeClass() and tearDownAfterClass() template methods are called before the first test of the test case class is run and after the last test of the test case class is run, respectively. 

The correlation between atoum namings and PHPUnit namings is described by this table:

| atoum | PHPUnit |
|-|-|
| `setUp` | `setUpBeforeClass` |
| `tearDown` | `tearDownAfterClass` |
| `beforeTestMethod` | `setUp` |
| `afterTestMethod` | `tearDown` |

Thus, the `phpunit-extension` needs to override the `call*` methods to map atoum namings into PHPUnit namings. The virtual hooks are a really great solution, but they must have the protected visibility instead of being private.

**Comment by jubianchi** [comment]
good for me!

**Comment by Hywan** [comment]
I have included #773 is this branch too.

**Comment by jubianchi** [comment]
@Hywan something wen wrong while rebasing/merging this branch.

**Comment by Hywan** [comment]
@jubianchi What's wrong? I just merged `master` once again to be up-to-date.

**Comment by jubianchi** [comment]
@Hywan there should be only one commit in this PR.

**Comment by Hywan** [comment]
Rebased on top of `master`.

**Comment by Hywan** [comment]
@jubianchi It guess it's time to merge this PR and make a new release of atoum :-).

**Comment by Hywan** [comment]
Rebased on top of master.

**Comment by Hywan** [comment]
Travis builds are failing because of `atoum/reports-extension` because of Symfony.

**Comment by Grummfy** [comment]
so, why it's not merged, seems ready

**Comment by Grummfy** [comment]
@Hywan I rerun the build seems working now
ready to merge and release?

**Comment by Grummfy** [comment]
I think once this conflict is resolve, we can merge to master

**Comment by Grummfy** [comment]
@Hywan & @jubianchi  if it's ready, except resolving the conflict, we can merge, can you just confirm it?

File: classes/test.php [hunk 2]
```diff
@@ -1748,14 +1753,14 @@ function ($value) use (& $methodName) {
...
                             switch ($value[0]) {
-                                case '<':
-                                case '<=':
-                                case '=':
-                                case '==':
-                                case '>=':
-                                case '>':
-                                    $operator = $value[0];
-                            }
+                            case '<':
```
Comment (trasher) [inline classes/test.php]: indentation seems incorrect

**Review by cedric-anne — APPROVED** [review]
Seems OK. Should be rebased before beiing merged.

**Pull request merged**